torch/csrc/autograd/python_variable.cpp

```diff
@@ -186,6 +187,8 @@ PyObject *THPVariable_pynew(PyTypeObject *type, PyObject *args, PyObject *kwds)
     var = make_variable(torch::createTensor(data), requires_grad, is_volatile);
   }
 
+  var.pretty_name() = std::string(pretty_name ? pretty_name : "");
```

Comment:
How about `if (pretty_name) var.pretty_name() = pretty_name;`? Might be faster in case `pretty_name` is NULL (which is most of the time) because it won't have to run `operator=` and additional constructor.